DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs

```diff
@@ -253,7 +255,21 @@ await _virtualDeviceStorage.AddOrUpdateDeviceAsync(new InitialDeviceConfig()
                 DeviceSchemaHelper.InitializeSystemProperties(device, null);
             }
 
-            return await _deviceRegistryCrudRepository.UpdateDeviceAsync(device);
+            // Merge device back to existing so we don't drop missing data
+            ((JObject)existingDevice).Merge(device);
```

Comment:
If existing device isn't a JObject, this will throw an error. Please check to see if it is a JObject before doing this cast.